flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -188,7 +183,12 @@ abstract class PlannerBase(
                 s"${classOf[OverwritableTableSink].getSimpleName} but actually got " +
                 sink.getClass.getName)
           }
-          LogicalSink.create(input, sink, identifier.toString, table)
+          LogicalSink.create(
+            input,
+            sink,
+            identifier.toString,
+            table,
+            catalogSink.getStaticPartitions.toMap)
```

Comment:
why toMap here? `catalogSink.getStaticPartitions` is already a Map

Reply:
The default corresponding class is `mutable.Map` in Scala for `java.util.Map` in Java.
But the default `Map` in Scala is immutable map. We should use immutable map too. So need `toMap` here.